omc3/global_correction.py

```diff
@@ -0,0 +1,356 @@
+                         help="Optimization method to use.", )
+    params.add_parameter(name="svd_cut",
+                         type=float,
+                         default=CORRECTION_DEFAULTS["svd_cut"],
+                         help="Cutoff for small singular values of the pseudo inverse. "
+                              "(Method: 'pinv')Singular values smaller than "
+                              "rcond*largest_singular_value are set to zero", )
+    params.add_parameter(name="n_correctors",
+                         type=int,
+                         help="Maximum number of correctors to use. (Method: 'omp')")
+    params.add_parameter(name="max_iter",
+                         type=int,
+                         default=CORRECTION_DEFAULTS["max_iter"],
+                         help="Maximum number of correction re-iterations to perform. "
+                              "A value of `0` means the correction is calculated once.", )
+    params.add_parameter(name="use_errorbars",
+                         action="store_true",
+                         help="Take into account the measured errorbars in the correction.", )
+    params.add_parameter(name="update_response",
+                         action="store_true",
+                         help="Update the (analytical) response per iteration.", )
+    return params
+
+
+@entrypoint(correction_params())
+def global_correction_entrypoint(opt: DotDict, accel_opt) -> None:
+    """Do the global correction. Iteratively."""
+    LOG.info("Starting Iterative Global Correction.")
+    opt = _check_opt_add_dicts(opt)
+    opt = _add_hardcoded_paths(opt)
+    opt.output_dir.mkdir(parents=True, exist_ok=True)
+    accel_inst = manager.get_accelerator(accel_opt)
+    handler.correct(accel_inst, opt)
+
+
+def _check_opt_add_dicts(opt: dict) -> dict:  # acts inplace...
+    """ Check on options and put in missing values """
+    def_dict = _get_default_values()
+    opt.optics_params = [param.replace(f"{BETABEAT}", f"{BETA}") for param in opt.optics_params]
+    for key in ("modelcut", "errorcut", "weights"):
+        if opt[key] is None:
+            opt[key] = [def_dict[key][p] for p in opt.optics_params]
+        elif len(opt[key]) != len(opt.optics_params):
+            raise AttributeError(f"Length of {key} is not the same as of the optical parameters!")
+        opt[key] = dict(zip(opt.optics_params, opt[key]))
+    opt.meas_dir = Path(opt.meas_dir)
+    opt.output_dir = Path(opt.output_dir)
+    return opt
+
+
+def _add_hardcoded_paths(opt: DotDict) -> DotDict:  # acts inplace...
+    opt.change_params_path = opt.output_dir / f"{opt.output_filename}.madx"
+    opt.change_params_correct_path = opt.output_dir / f"{opt.output_filename}_correct.madx"
+    opt.knob_path = opt.output_dir / f"{opt.output_filename}.tfs"
+    return opt
+
+
+# Define functions here, to new optics params
+def _get_default_values() -> Dict[str, Dict[str, float]]:
```

Comment:
They are more or less empirical values. I don't think there is a good "why". What do you think @lmalina @mihofer @tpersson @rogeliotomas

Reply:
Yeah the default values are more based on experience. I would adjust the weights to 1 for f1001R and f1010R. I would also give weight to normalized dispersion, when it comes to betx and bety they should also have some default values in case they are only taken from the k-mod.

Reply:
@JoschD I think these changes are all we need before merging.